samples/go/csv/read.go

```diff
@@ -178,11 +194,10 @@ func ReadToMap(r *csv.Reader, structName string, headersRaw []string, primaryKey
 		}
 
 		fields := readFieldsFromRow(row, headersRaw, fieldOrder, kindMap)
-		kvChan <- fields[fieldOrder[pkIndices[0]]]
-		kvChan <- types.NewStructWithType(t, fields)
+		graphKeys, mapKey := primaryKeyFromFields(fields, fieldOrder, pkIndices)
+		gb.MapSet(graphKeys, mapKey, types.NewStructWithType(t, fields))
 	}
-	close(kvChan)
-	return <-mapChan
+	return gb.Build().(types.Map)
```

Comment:
Very clean, nice work with GraphBuilder! Just a few suggestions included above. I am assuming the tests still all work correctly. Probably worthwhile to include tests for primaryKeyFromFields too. Back to you.

@kalman Do we have perf results for a nested map currently so we can determine speed improvements?